### PR TITLE
mon: compensate for not having STDLIB_MAP_SPLICING

### PR DESCRIPTION
**Pull request by wjwithagen**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -8685,7 +8685,13 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
      mempool::osdmap::map<string,map<string,string>> old_ec_profiles =
        osdmap.get_erasure_code_profiles();
      auto ec_profiles = pending_inc.get_erasure_code_profiles();
-     ec_profiles.merge(old_ec_profiles);
+#ifdef HAVE_STDLIB_MAP_SPLICING
+     ec_profiles.insert(old_ec_profiles);
```
Comment (jecluis) [inline src/mon/OSDMonitor.cc]: shouldn't this have been `ec.profiles.merge()`?

Comment (wjwithagen) [reply src/mon/OSDMonitor.cc]: @jecluis 
Sharp eye...
Thanx

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -8685,7 +8685,13 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
      mempool::osdmap::map<string,map<string,string>> old_ec_profiles =
        osdmap.get_erasure_code_profiles();
      auto ec_profiles = pending_inc.get_erasure_code_profiles();
+#ifdef HAVE_STDLIB_MAP_SPLICING
      ec_profiles.merge(old_ec_profiles);
+#else
+     for (auto &i: old_ec_profiles) {
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: i'd suggest follow other places where we workaround missing `std::map.merge()` , like:

```c++
ec_profiles.insert(make_move_iterator(begin(old_ec_profiles)),
                   make_move_iterator(end(old_ec_profiles)));
```

**Comment by tchaikov** [comment]
@jecluis we could, by setting up a jenkins job to build ceph with clang 7. see https://libcxx.llvm.org/cxx1z_status.html ("Splicing Maps and Sets" was implemented in libc++ 8).

**Comment by jecluis** [comment]
@tchaikov I'm not sure if that's worth the effort; any idea who would have an informed opinion on this?

**Comment by wjwithagen** [comment]
@jecluis 
I, for one, would really like that, because it would alert others to possible porting problems.
The option of doing this, now and then is raised, but also fades out.

Lots of the PR's I submit fall exactly in this category, and it is that I do not have enough compute power to keep up with compiling every PR. But otherwise I would off that as inclusion for a PR PASS requirement.

File: src/mon/OSDMonitor.cc (hunk 2 above)
Comment (tchaikov) [reply src/mon/OSDMonitor.cc]: @wjwithagen ping?

**Pull request merged**